mal/useranimelist.go

```diff
@@ -196,6 +198,32 @@ type Comments string
 func (c Comments) updateMyAnimeListStatusApply(v *url.Values) { v.Set("comments", string(c)) }
 func (c Comments) updateMyMangaListStatusApply(v *url.Values) { v.Set("comments", string(c)) }
 
+// StartDate is an option that allows to update the start date of anime and manga
+// in the user's list.
+type StartDate time.Time
+
+// YYYYMMDD is the layout string for the start date and finish date of anime
+// and manga in the user's list.
+const YYYYMMDD = "2006-01-02"
```

Comment:
We need to also be able to remove the values of start and finish dates.
A good way to achieve that is to use the zero value of `time.Time` as we cannot pass `nil`.

Example:

```go
s, _, err := c.Anime.UpdateMyListStatus(ctx, 967,
	mal.AnimeStatusWatching,
	mal.StartDate(time.Date(2022, 02, 20, 0, 0, 0, 0, time.UTC)),
	mal.FinishDate(time.Time{}), // Remove the value of finish date
)
```
Therefore, I am suggest to replace the constant with a small function that handles the zero value.
```suggestion
func formatMALDate(d time.Time) string {
	if d.IsZero() {
		return ""
	}
	return d.Format("2006-01-02")
}
```